authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,153 @@
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+from django.urls import reverse
+from authors.apps.articles.models import Article, Ratings
+from authors.apps.articles.tests.test_data import (
+    valid_article, valid_article_2,
+    valid_rating, slug, invalid_rating, valid_rating_2, 
+    invalid_rating2, invalid_integer, 
+    signup_and_login_user
+    )
+from authors.apps.authentication.models import User
+from authors.apps.authentication.tests.test_data import (
+    valid_user, valid_user2
+    )
+from authors.apps.authentication.tests.app_urls import (
+    register_url, login_url, activate_url, user_action_url
+    )
+
+
+class RateArticleAPIViewTestCase(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        response = self.client.post(
+            register_url, {"user": valid_user}, format="json")
+
+        self.token = response.data["data"]["token"]
+
+        self.client.get(f"{activate_url}?token={self.token}", format='json')
+
+        response = self.client.post(
+            login_url, {"user": valid_user}, format='json')
+
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+
```

Comment:
Hi @NicholusMuwonge, there is an easier and neater way to authenticate a user and you used it in the function you wrote to  `signup_and_login_user`.  Is there a particular reason you decided to do the authentication this way in the `setUp` method?

Reply:
Yeah, at first, i was getting errors with the one in the ```signup_and_login_user```

Reply:
but I have updated